### PR TITLE
Fix #1040 : add options to be able to sort APIs and operations

### LINKED ISSUE #1040: Sorter option with 2.1.8-M1
Hi,
I'm trying to migrate my apis to swagger 2.0 and I can't sort APIs in a tag group on the UI like before.
Sorter option have no effect (I tried "alpha" and "method").
I take a look in the code, and I see nothing that call sorter method. This option is no longer available in M1?
Thanks.

### PR DESCRIPTION
**Pull request by julien-maurel**

### CODE REVIEW

**Comment by fehguy** [comment]
@mohsen1 can you see if this can be merged?

**Comment by julien-maurel** [comment]
No, travis faild... I pushed a fix for jshint, build is waiting.

**Comment by mohsen1** [comment]
Please delete ` src/main/javascript/view/MainView.js~` file which I think is a leftover

File: src/main/javascript/view/MainView.js [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
   initialize: function(opts){
+    var sorterName, sorter, key, value;
     opts = opts || {};
 
     this.router = opts.router;
 
+    // Sort APIs
+    if (opts.swaggerOptions.apisSorter) {
+      sorterName = opts.swaggerOptions.apisSorter;
+      if (typeof sorterName === 'function') {
```
Comment (mohsen1) [inline src/main/javascript/view/MainView.js]: This API is a little confusing. `sorterName` can be both string and function. Can we keep `sorterName` to be always a string and have another field for the function variant? Like `sorterFn`.

File: src/main/javascript/view/MainView.js [hunk 2]
```diff
@@ -1,21 +1,44 @@
...
+        sorter = sorterName;
+      } else {
+        sorter = this.apisSorters[sorterName];
```
Comment (mohsen1) [inline src/main/javascript/view/MainView.js]: Before using `sorter` as a sort comparer we need to make sure it's a function. 

```js
if (_.isFunction(sorter)) {
```

File: src/main/javascript/view/MainView.js [hunk 3]
```diff
@@ -1,21 +1,44 @@
...
+      }
+      this.model.apisArray.sort(sorter);
+    }
+    // Sort operations of each API
+    if (opts.swaggerOptions.operationsSorter) {
```
Comment (mohsen1) [inline src/main/javascript/view/MainView.js]: Same comments about API and type check applies to `operationsSorter` too.

**Comment by mohsen1** [comment]
This is very well written. There are some small issues I've commented. Please fix those and update README.md with documentations about this, then we can merge it.

File: src/main/javascript/view/MainView.js (hunk 1 above)
Comment (julien-maurel) [inline src/main/javascript/view/MainView.js]: sorterName is a shortcut to opts.swaggerOptions.apisSorter and I don't know if it is a string or a function at assignement. I'll rename this shortcut to sorterOption and after sorterFn will contain function

**Comment by julien-maurel** [comment]
Ok, I do changes and I push again. Let me know if something wrong.

**Comment by mohsen1** [comment]
Thanks @julien-maurel. This looks good to me. @fehguy please merge.

**Pull request merged**